polyfactory/factories/sqlalchemy_factory.py

```diff
@@ -174,5 +57,20 @@ def is_supported_type(cls, value: Any) -> TypeGuard[type[T]]:
 
     @classmethod
     def get_model_fields(cls) -> list[FieldMeta]:
-        # TODO
-        raise NotImplementedError()
+        fields_meta: list[FieldMeta] = []
+
+        types_override = cls.get_sqlalchemy_types()
+        columns = cls.__model__.__table__.columns
+        for name, column in columns.items():
+            annotation: type = type(column.type) if type(column.type) in types_override else column.type.python_type
```

Comment:
I think the handling of the type of the items in the list if the type is `ARRAY` is missing. If I'm not mistaken, `column.type.python_type` will only return `list` and not `List[str]` or `List[int]` or whatever be the case.

Reply:
Good point. I have updated to extract origin for ARRAY SQLAlchemy type here.

Note I think a similar issue exists for tuple type but I wasn't able to add better support for this easily.